client/src/pages/Listing/Listing.tsx

```diff
@@ -1,77 +1,91 @@
 import { Box, Grid, IconButton, InputAdornment, TextField, Typography } from '@material-ui/core';
+import CheckIcon from '@material-ui/icons/Check';
 import ClearIcon from '@material-ui/icons/Clear';
 import SearchIcon from '@material-ui/icons/Search';
 import clsx from 'clsx';
+import { useEffect, useRef, useState } from 'react';
 import { CustomButton } from '../../components/Button/CustomButton';
 import NavBar from '../../components/NavBar/NavBar';
+import { useSnackBar } from '../../context/useSnackbarContext';
+import { listProfiles } from '../../helpers/APICalls/listProfiles';
+import { Profile } from '../../interface/Profile';
 import { ListingItem, rate } from './ListingItem';
 import useStyles from './useStyles';
 
-const sitters = [
-  {
-    id: 1,
-    fullname: 'Johnny Yad',
-    img: 'https://i.pravatar.cc/300',
-    slogan: 'Professional dog trainer',
-    description: 'I would like to work with your dog.',
-    location: 'New Yourk',
-    price: 80,
-    rating: 5,
-  },
-  {
-    id: 2,
-    fullname: 'Johnny Yad',
-    img: 'https://i.pravatar.cc/300',
-    slogan: 'Professional dog trainer',
-    description: 'I would like to work with your dog.',
-    location: 'New Yourk',
-    price: 80,
-    rating: 3,
-  },
-  {
-    id: 3,
-    fullname: 'Johnny Yad',
-    img: 'https://i.pravatar.cc/300',
-    slogan: 'Professional dog trainer',
-    description: 'I would like to work with your dog.',
-    location: 'Paris',
-    price: 180,
-    rating: 0,
-  },
-  {
-    id: 4,
-    fullname: 'Johnny Yad',
-    img: 'https://i.pravatar.cc/300',
-    slogan: 'Professional dog trainer',
-    description: 'I would like to work with your dog.',
-    location: 'New Yourk',
-    price: 40,
-    rating: 4,
-  },
-  {
-    id: 5,
-    fullname: 'Johnny Yad',
-    img: 'https://i.pravatar.cc/300',
-    slogan: 'Professional dog trainer',
-    description: 'I would like to work with your dog.',
-    location: 'New Yourk',
-    price: 80,
-    rating: 4,
-  },
-  {
-    id: 6,
-    fullname: 'Johnny Yad',
-    img: 'https://i.pravatar.cc/300',
-    slogan: 'Professional dog trainer',
-    description: 'I would like to work with your dog.',
-    location: 'New Yourk',
-    price: 80,
-    rating: 4,
-  },
-];
-
 const Listing = (): JSX.Element => {
   const { root, title, search, dateInOff, iconStyle } = useStyles();
+  const TODAY = new Date().getDay();
+  const [loading, setLoading] = useState<undefined | boolean>(undefined);
+  const [filter, setFilter] = useState<boolean>(false);
+  const { updateSnackBarMessage } = useSnackBar();
+  const [profiles, setProfiles] = useState<Profile[]>([]);
+  const [dateIn, setDateIn] = useState<number>(TODAY);
+  const [dateOff, setDateOff] = useState<number>(TODAY);
+  const [searchString, setSearchString] = useState<string>('');
+  const dateInRef = useRef<any>();
+  const dateOffRef = useRef<any>();
+  const searchRef = useRef<any>();
+
+  const handleFilterChange = () => {
+    setFilter(!filter);
```

Comment:
```suggestion
    setFilter(prevFilter => !prevFilter);
```

It is a good practice to pass a function that takes the current state as the first argument!

Reply:
done ✔️